control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go

```diff
@@ -140,150 +140,167 @@ func setConditionByType(conditions *[]hyperv1.HostedControlPlaneCondition, condi
 	}
 }
 
-func (r *HostedControlPlaneReconciler) setAvailableCondition(ctx context.Context, hostedControlPlane *hyperv1.HostedControlPlane, oldStatus *hyperv1.HostedControlPlaneStatus,
-	status hyperv1.ConditionStatus, reason, message string, result ctrl.Result, err error) (ctrl.Result, error) {
-	conditions := &hostedControlPlane.Status.Conditions
-
-	setConditionByType(conditions, hyperv1.Available, status, reason, message)
-
-	// Sync status.ready with Available condition
-	condition := getConditionByType(*conditions, hyperv1.Available)
-	if condition != nil && condition.Status == hyperv1.ConditionTrue {
-		hostedControlPlane.Status.Ready = true
-	} else {
-		hostedControlPlane.Status.Ready = false
-	}
-
-	if reflect.DeepEqual(oldStatus, hostedControlPlane.Status) {
-		// No change to status, nothing to sync
-		return result, err
-	}
-
-	// Check for changed conditions to update LastTransitionTime
-	for k, condition := range *conditions {
-		oldCondition := getConditionByType(oldStatus.Conditions, condition.Type)
-		if oldCondition == nil || *oldCondition != condition {
-			(*conditions)[k].LastTransitionTime = metav1.Now()
-		}
-	}
-
-	if updateErr := r.Status().Update(ctx, hostedControlPlane); updateErr != nil {
-		r.Log.Error(err, "failed to update status")
-		result.Requeue = true
-	}
-
-	return result, err
-}
-
-func (r *HostedControlPlaneReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+func (r *HostedControlPlaneReconciler) Reconcile(ctx context.Context, req ctrl.Request) (result ctrl.Result, reconcileErr error) {
 	r.Log = ctrl.LoggerFrom(ctx)
 	r.Log.Info("Reconciling")
 
+	defer func() {
```

Comment:
This will be be duplicated by controller runtime logging https://github.com/kubernetes-sigs/controller-runtime/pull/1245/files

I'd rather avoid the defer call and `return fmt.Errorf("failed to get whatever: %w", err)` explicitly in the statements below